features/networking/service.feature

```diff
@@ -153,3 +153,325 @@ Feature: Service related networking scenarios
     Then the step should succeed
     And the output should contain:
       | 8080 |
+
+  # @author weliang@redhat.com
+  # @case_id OCP-24668
+  Scenario: externalIP defined in service but no sepc.externalIP defined	
+    Given I have a project 
+    # Create a service with a externalIP
+    When I run the :create client command with:
+      | f | https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/externalip_service1.json | 
+    Then the step should fail
+    Then the output should contain "externalIPs have been disabled"
+
+  # @author weliang@redhat.com
+  # @case_id OCP-24669
+  @admin
+  @destructive
+  Scenario: externalIP defined in service with set ExternalIP in allowedCIDRs	 
+    Given I store all worker nodes to the :nodes clipboard
+    And the Internal IP of node "<%= cb.nodes[0].name %>" is stored in the :hostip clipboard
+   
+    # Create additional network through CNO
+    Given as admin I successfully merge patch resource "networks.config.openshift.io/cluster" with:
+      | {"spec":{"externalIP":{"policy":{"allowedCIDRs":["<%= cb.hostip %>/24"]}}}} |
+    And I wait up to 10 seconds for the steps to pass:
+    """
+    When I run the :get admin command with:
+      | resource      | networks.config.openshift.io                 |
+      | resource_name | cluster                                      |
+      | template      | {{{{.spec.externalIP.policy.allowedCIDRs}}}} |
+      | o             | yaml                                         |
+    And the output should contain "<%= cb.hostip %>"
+    """
+    #Clean-up required to erase above externalIP policy after testing done
+    Given I register clean-up steps:
+    """
+    Given as admin I successfully merge patch resource "networks.config.openshift.io/cluster" with:
+      | {"spec":{"externalIP":{"policy":{"allowedCIDRs": null}}}}    |
+    """
+
+    # Create a svc with externalIP
+    Given I have a project 
+    And I wait up to 400 seconds for the steps to pass:
+    """
+    When I run oc create over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/externalip_service1.json" replacing paths:
+      | ["spec"]["externalIPs"][0]  | <%= cb.hostip %> |
+    Then the step should succeed
+    """ 
+    Then I wait up to 10 seconds for the steps to pass:
+    """
+    When I run the :get client command with:
```

Comment:
@weliang1 we should try to avoid calling `:get` directly and use object notation when possible.  If it's not there we can add the support in the framework....if your case it's probably something like `service('service-unsecure')`

Reply:
@pruan-rht sorry, I not get it your point here, could you point me any existing script samples wrote as you suggested above? Thanks!

Reply:
@pruan-rht No need those :get steps, remove them from the scripts.